modules/indexing/src/test/java/org/apache/ignite/cache/query/ThinClientIndexQueryTest.java

```diff
@@ -224,14 +227,25 @@ public void testPageSize() {
             for (int pageSize: F.asList(1, 10, 100, 1000, 10_000)) {
                 idxQry.setPageSize(pageSize);
 
-                TestRecordingCommunicationSpi.spi(grid(0)).record(GridQueryNextPageRequest.class);
+                for (int i = 0; i < NODES; i++)
+                    TestRecordingCommunicationSpi.spi(grid(i)).record(GridCacheQueryRequest.class);
 
                 assertClientQuery(cache, NULLS_CNT, CNT, idxQry);
 
-                List<Object> reqs = TestRecordingCommunicationSpi.spi(grid(0)).recordedMessages(true);
+                List<Object> reqs = new ArrayList<>();
+
+                for (int i = 0; i < NODES; i++) {
+                    reqs.addAll(TestRecordingCommunicationSpi.spi(grid(i)).recordedMessages(true)
+                        .stream()
+                        .map(o -> (GridCacheQueryRequest)o)
+                        .filter(o -> o.type() == null)
```

Comment:
This filter doesn't affect test results

Reply:
The processing of messages has been changed altogether as another check for `GridCacheQueryResponse` messages has been added.